File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        
        unordered_map<int, int> counts;

        for (int num : nums) {
            counts[num]++;
        }

        vector<pair<int, int>> sorted;

        for (pair<int, int> p : counts) {
            sorted.push_back({p.second, p.first});
        }

        sort(sorted.rbegin(), sorted.rend());

        vector<int> out;

        for( int i = 0; i < k; i++ ) {
            out.push_back(sorted[i].second);
        }

        return out;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: list[int]) -> list[list[int]]:
        
        out = []
        
        nums.sort()

        for i, a in enumerate(nums):

            # Skip duplicates for first pointer
            if i > 0 and a == nums[i - 1]:
                continue

            l = i + 1
            r = len(nums) - 1

            while l < r:
                ts = a + nums[l] + nums[r]

                if ts < 0:
                    l += 1
                elif ts > 0:
                    r -= 1
                else:
                    out.append([a, nums[l], nums[r]])
                    l += 1

                    while l < r and nums[l] == nums[l - 1]:
                        l += 1
        
        return out
